plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -181,41 +181,58 @@ export const SearchResult = ({ searchQuery }: SearchResultProps) => {
   }
 
   const resetFilters = () => {
-    setFilters({
+    setSelectedFilters({
       selected: 'All',
       checked: [],
     });
   };
 
   const updateSelected = (filter: string) => {
-    setFilters(prevState => ({
+    setSelectedFilters(prevState => ({
       ...prevState,
       selected: filter,
     }));
   };
 
   const updateChecked = (filter: string) => {
-    if (filters.checked.includes(filter)) {
-      setFilters(prevState => ({
+    if (selectedFilters.checked.includes(filter)) {
+      setSelectedFilters(prevState => ({
         ...prevState,
         checked: prevState.checked.filter(item => item !== filter),
       }));
       return;
     }
 
-    setFilters(prevState => ({
+    setSelectedFilters(prevState => ({
       ...prevState,
       checked: [...prevState.checked, filter],
     }));
   };
 
+  const filterOptions = results.reduce(
+    (acc, curr) => {
+      if (curr.kind && acc.kind.indexOf(curr.kind) < 0) {
+        acc.kind.push(curr.kind);
+      }
+      if (curr.lifecycle && acc.lifecycle.indexOf(curr.lifecycle) < 0) {
+        acc.lifecycle.push(curr.lifecycle);
+      }
+      return acc;
+    },
+    {
+      kind: [] as Array<string>,
+      lifecycle: [] as Array<string>,
+    },
+  );
+
   return (
     <>
       <Grid container>
         {showFilters && (
           <Grid item xs={3}>
             <Filters
-              filters={filters}
+              filters={selectedFilters}
```

Comment:
filters can be replaced with selectedFilters as well and also in the Filters component (see my comment above)

Reply:
My thinking was that `filters` in the Filters component will be considered as selected filters by default, since we are passing them to the Filters component. Do you think it might be confusing?

Reply:
@magneticz now when we have filterOptions it could be, but I don't want to block this on that so lets leave it as it is for now! You might want to do a rebase on top of master though, we have merged some new stuff!